WeightProducer/python/MCSampleValues.py

```diff
@@ -50,506 +68,1240 @@ def get_year(self, mcVersion):
-            "CrossSection" : XSValues(XS_13TeV=43.5543),
-        },
-        "ZJetsToNuNu_Zpt-200toInf" : {
-            "CrossSection" : XSValues(XS_13TeV=5.28654),
+            "CrossSection" : XSValues(
+                XS_2016=44.27, XSSource_2016="XSDB (LO)",
+                XS_2017=40.65, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=40.46, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "ST_s-channel_4f_hadronicDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=6.96),
+            "CrossSection" : XSValues(
+                XS_2017=11.24, XSSource_2017="GenXSecAnalyzer",
+                XS_2018=11.24, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
-        # single top: NoFullyHadronicDecays xsec scaled by BF for non-fully-hadronic (1-(1-3*0.108)^2)
         "ST_s-channel_4f_leptonDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=3.34),
+            "CrossSection" : XSValues(
+                XS_2016=3.365, XSSource_2016="XSDB (unknown)",
+                XS_2017=3.74,  XSSource_2017="XSDB (unknown)",
+                XS_2018=3.740, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "ST_s-channel_4f_InclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=10.32),
-        },
-        "ST_t-channel_top_4f_inclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=136.02),
+            "CrossSection" : XSValues(
+                XS_13TeV=10.12, XSSource_13TeV="XSDB (unknown)",
+            ),
         },
         "ST_t-channel_top_4f_InclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=136.02),
-        },
-        "ST_t-channel_antitop_4f_inclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=80.95),
+            "CrossSection" : XSValues(
+                XS_2016=136.02, XSSource_2016="Unknown (GenXSecAnalyzer/XSDB return 0.0)",
+                XS_2017=113.3,  XSSource_2017="XSDB (NLO)",
+                XS_2018=115.3,  XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "ST_t-channel_antitop_4f_InclusiveDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=80.95),
-        },
-        "ST_tW_antitop_5f_NoFullyHadronicDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=19.4674),
+            "CrossSection" : XSValues(
+                XS_2016=80.95, XSSource_2016="Unknown (GenXSecAnalyzer/XSDB return 0.0)",
+                XS_2017=67.91, XSSource_2017="XSDB (NLO)",
+                XS_2018=69.09, XSSource_2018="GenXSecAnalyzer",
+            ),
         },
         "ST_tW_top_5f_NoFullyHadronicDecays" : {
-            "CrossSection" : XSValues(XS_13TeV=19.4674),
+            "CrossSection" : XSValues(
+                XS_2016=38.09, XSSource_2016="XSDB (NLO)",
+                XS_2017=34.91, XSSource_2017="XSDB (NLO)",
+                XS_2018=34.91, XSSource_2018="GenXSecAnalyzer",
+            ),
+            "BranchingRatio" : BRValues(
```

Comment:
make this a single BR_13TeV (also below)

Reply:
Changed